src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java

```diff
@@ -399,61 +351,115 @@ public boolean setLocation(Location<World> location, boolean forced) {
         for (net.minecraft.entity.Entity passenger : passengers) {
             passenger.startRiding(thisEntity, true);
         }
-
-        return true;
     }
 
-    @Override
-    public void setLocationAndRotation(Location<World> location, Vector3d rotation, EnumSet<RelativePositions> relativePositions) {
-        setLocationAndRotation(location, rotation, relativePositions, false);
-    }
+    // for sponge internal use only
+    @SuppressWarnings("unchecked")
+    private boolean teleportTo(Location<World> location) {
+        final MinecraftServer server = (MinecraftServer) Sponge.getServer();
+        final net.minecraft.entity.Entity this$ = (net.minecraft.entity.Entity) (Object) this;
```

Comment:
Don't like dollars in variable names. Use `entity`, `self` or `mcEntity` for example